src/pixelator/analysis/polarization/__init__.py

```diff
@@ -32,28 +30,29 @@
 logger = logging.getLogger(__name__)
 
 
-def morans_autocorr(w: np.ndarray, y: np.ndarray) -> Any:
+def morans_autocorr(w: np.ndarray, y: np.ndarray, permutations: int) -> Any:
     """
-    A helper function that computes the Moran's I autocorrelation statistics
+    A function that computes the Moran's I autocorrelation statistics
     for the given spatial weights (w) and target variable (y). The target
     variable can be the counts of a specific antibody so the Moran's I value
     could indicate if the antibody has a localized spatial pattern. The function
     returns an object with the Moran's statistics (I, p_rand and z_rand).
     :param w: the weights matrix (connectivity matrix)
     :param y: the counts vector
+    :param permutations: the number of permutations for simulated Z-score (z_sim) 
+                         estimation (if permutations>0)
     :returns: the Moran's statistics
     """
     # Default Moran's transformation is row-standardized "r".
     # https://github.com/pysal/esda/blob/main/esda/moran.py
-    return Moran(y, w, permutations=None)
+    return Moran(y, w, permutations=permutations)
 
 
 def polarization_scores_component(
     graph: Graph,
     component_id: str,
-    normalization: Literal["raw", "clr", "denoise"] = "clr",
-    antibody_control: Optional[List[str]] = None,
-    binarization: bool = False,
+    normalization: Literal["raw", "clr"] = "clr",
+    permutations: int = None,
```

Comment:
Are you sure you want `None` as a default here?

Reply:
I was thinking we can leave it as `None` for the time being, and change default to 999 if we choose to go with permuted scores as a default. What do you think?

Reply:
I guess you want to have here the default number of permutations, like in the colocalization scoring part. Then, we report all the different scores at the same time. But when we are going to try doing permutations, the defaults should be set.